hsmodels/schemas/rdf/validators.py

```diff
@@ -24,6 +24,15 @@ def rdf_parse_identifier(cls, value):
     return value
 
 
+def subjects_constraint(cls, subjects):
+    """Removes empty/None and duplicates"""
+    validated = []
+    for subject in subjects:
+        if subject and subject not in validated:
+            validated.append(subject)
+    return validated
```

Comment:
@sblack-usu it'd be better to trim each subject before line 31. Also, it is probably better to convert subjects into lower case to account for same subjects with only case difference? Just some minor edge cases for consideration, but not a big deal at all.

Reply:
I can add the trim.

Reply:
I don't think we should be lowercasing subjects, hydroshare does not do that today.  I think discover might do that when indexing.

Reply:
OK. Sounds good.

Reply:
This is really nitpicking, but it'd be better to strip() the subject before checking it on line 31. I am approving the PR now, feel free to merge it or make this small change then merge it.

Reply:
It's a valid critique, done.